packages/selinux-policy/systems.cil

```diff
@@ -21,5 +21,7 @@
 (classmapping systems use (key (all)))
 (classmapping systems use (lockdown (all)))
 (classmapping systems use (perf_event (all)))
+(classmapping systems use (io_uring (all)))
+(classmapping systems use (user_namespace (all)))
```

Comment:
I'm sad about this, but I don't know what compatibility-breaking gremlins would be unleashed by restricting user namespace creation to privileged pods and tasks.

Reply:
Would right now not be the time if we want to (try to) lock this down? In the default configuration, user namespaces remain turned off on kernels 5.10, 5.15, and 6.1. Only kernel 6.1 make user namespace creation controllable via SELinux. The change in policy could be safely tried for the new variant (and weakened if need be) for users turning on user namespaces.

Reply:
Effectively today we are defaulting to "all" and can't change that default in the SELinux policy without some way to opt into the old behavior.

To do this I think we'd want a setting, like `settings.kernel.allow-user-namespaces`, with two or three values:
* "all" - all processes are allowed to create user namespaces
* "privileged" - privileged processes are allowed to create user namespaces
* "trusted" - trusted processes are allowed to use create namespaces

Then we would conditionally load an SELinux module based on which field was set.

We would then set `user.max_user_namespaces` up from zero by default and rely on the setting to restrict or allow access as needed.

I don't think it has to happen now but it would be a nice way to let people opt into more fine-grained access to user namespaces.

Reply:
The mechanism you propose is much more polished than what I had in mind for now. Since the 6.1 kernel would only be included in new variants, and the SELinux policy change hence constrained to those, I saw this as a chance to try to this admittedly incompatible change. If there were users opting into user namespaces and using them from unprivileged pods, we'd hear during their upgrade testing. From your response I take it you're against such changes even across variant upgrades, though, which is fine--the proposal was somewhat brisk. :-)